pkg/operator/controller/sync-http-error-code-configmap/httperrorcode_configmap.go

```diff
@@ -0,0 +1,115 @@
+	haveCMInOpenShiftConfig, currentConfigMapInOpenShiftConfig, err := r.currentHttpErrorCodeConfigMap(ic, "openshift-config")
+	log.Info(fmt.Sprintf(" currentConfigMapInOpenShiftConfig %v", currentConfigMapInOpenShiftConfig))
+	if err != nil {
+		return false, nil, err
+	}
+	haveCMInOpenShiftIngress, currentConfigMapInOpenShiftIngress, err := r.currentHttpErrorCodeConfigMap(ic, "openshift-ingress")
+	if err != nil {
+		return false, nil, err
+	}
+	switch {
+	case ic.Spec.HttpErrorCodePages.Name != "" && !haveCMInOpenShiftConfig && !haveCMInOpenShiftIngress:
+		return false, nil, nil
+	case ic.Spec.HttpErrorCodePages.Name != "" && !haveCMInOpenShiftConfig && haveCMInOpenShiftIngress:
+		if err := r.client.Delete(context.TODO(), currentConfigMapInOpenShiftIngress); err != nil {
+			if !errors.IsNotFound(err) {
+				return true, currentConfigMapInOpenShiftIngress, fmt.Errorf("failed to delete configmap: %v", err)
+			}
+		} else {
+			log.Info("deleted configmap", "configmap", currentConfigMapInOpenShiftIngress)
+		}
+		return false, nil, nil
+	case ic.Spec.HttpErrorCodePages.Name != "" && haveCMInOpenShiftConfig && !haveCMInOpenShiftIngress:
+		_, currentConfigMapInOpenShiftIngress, err = desiredHttpErrorCodeConfigMap(currentConfigMapInOpenShiftConfig, deploymentRef)
+		if err := r.client.Create(context.TODO(), currentConfigMapInOpenShiftIngress); err != nil {
+			return false, nil, fmt.Errorf("failed to create configmap: %v", err)
+		}
+		log.Info("created configmap", "configmap", currentConfigMapInOpenShiftIngress)
+		return r.currentHttpErrorCodeConfigMap(ic, "openshift-ingress")
+	case ic.Spec.HttpErrorCodePages.Name != "" && haveCMInOpenShiftConfig && haveCMInOpenShiftIngress:
+		if updated, err := r.updateHttpErrorCodeConfigMap(currentConfigMapInOpenShiftIngress, currentConfigMapInOpenShiftConfig); err != nil {
+			return true, currentConfigMapInOpenShiftIngress, fmt.Errorf("failed to update configmap: %v", err)
+		} else if updated {
+			log.Info("updated configmap in openshift-ingress namespace", "configmap", currentConfigMapInOpenShiftIngress)
+			return r.currentHttpErrorCodeConfigMap(ic, "openshift-ingress")
+		}
+	}
+
+	return true, currentConfigMapInOpenShiftIngress, nil
+}
+
+// desiredRsyslogConfigMap returns the desired  httpErrorCodePageconfigmap.  Returns a
+// Boolean indicating whether a configmap is desired, as well as the configmap
+// if one is desired.
+func desiredHttpErrorCodeConfigMap(currentConfigMapInOpenShiftConfig *corev1.ConfigMap, deploymentRef metav1.OwnerReference) (bool, *corev1.ConfigMap, error) {
+	cm := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      currentConfigMapInOpenShiftConfig.Name,
+			Namespace: "openshift-ingress",
+		},
+		Data: currentConfigMapInOpenShiftConfig.Data,
+	}
+	cm.SetOwnerReferences([]metav1.OwnerReference{deploymentRef})
+	return true, &cm, nil
+}
+
+// CurrentHttpErrorCodeConfigMap returns the current httpErrorCodePage configmap.  Returns a
```

Comment:
duplicate function?

Reply:
Actually I need them for each for both controllers